manifests/init.pp

```diff
@@ -1,43 +1,42 @@
-# -- Class systemd
 # This module allows triggering systemd commands once for all modules
+#
+# @api public
+#
+# @param service_limits
+#   May be passed a resource hash suitable for passing directly into the
+#   ``create_resources()`` function as called on ``systemd::service_limits``
+#
+# @param manage_resolved
+#   Manage the systemd resolver
+#
+# @param resolved_ensure
+#   The state that the ``resolved`` service should be in
+#
+# @param manage_networkd
+#   Manage the systemd network daemon
+#
+# @param networkd_ensure
+#   The state that the ``networkd`` service should be in
+#
 class systemd (
-  $service_limits          = {},
-  Boolean $manage_resolved = true,
-  Boolean $manage_networkd = true,
+  Optional[Hash]            $service_limits  = undef,
```

Comment:
@raphink To the first comment: No, this is correct. You can do either of those cases in a valid manner.

`systemd::unit_file`` includes `systemd` so that it is independently feature complete and since this is the 90% use case of this module.